### PR TITLE
Add Troubleshooting documentation

### PR DESCRIPTION
**Pull request by JeffRMoore**
This pulls the really fantastic [Common Problems](https://github.com/koajs/koa/wiki/Common-Problems) wiki page into the main documentation as a troubleshooting guide to give it greater exposure.  I also linked to it from the main read me for the same reason.

I generalized the headings for the two 404 causes after surveying closed tickets to give more surface area in the document for problem diagnosis.

I added an example of a missing `await`, also based on survey of closed tickets.

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+### Solution
+
+The solution for this is rather easy, simply add `return next()` to the end of your helper middleware.
+
+```js
+router.use(function (ctx, next) {
+  if (ctx.query.id) {
+    ctx.state.id = parseInt(ctx.query.id);
+  }
+  return next();
```
Comment (tj) [inline docs/troubleshooting.md]: does the `return` actually have significance? I forget what we did with Koa 2.x now haha..

Comment (fl0w) [inline docs/troubleshooting.md]: @tj yes, it needs to return the promise :)

**Comment by jonathanong** [comment]
awesome

**Comment by PlasmaPower** [comment]
Should we remove the wiki page then? I don't want the pages diverging.

**Comment by jonathanong** [comment]
https://github.com/koajs/koa/wiki? sure

**Pull request merged**